diff --git a/password_policy.module b/password_policy.module
index 87f15021be1824e1a0a88b689fbe7d69759dc867..19c86db0259a6832b3429b11b7a42dd156ef942b 100644
--- a/password_policy.module
+++ b/password_policy.module
@@ -67,9 +67,9 @@ function password_policy_form_user_form_alter(&$form, FormStateInterface $form_s
   // Hide password reset field if no access.
   $account = \Drupal::currentUser();
   if (!$account->hasPermission('manage password reset')) {
-    $form['field_last_password_reset']['#access'] = FALSE;
-    $form['field_password_expiration']['#access'] = FALSE;
-    $form['field_pending_expire_sent']['#access'] = FALSE;
+    $form['field_last_password_reset']['#type'] = 'hidden';
+    $form['field_password_expiration']['#type'] = 'hidden';
+    $form['field_pending_expire_sent']['#type'] = 'hidden';
   }

   // Load form if relevant.