diff --git a/password_policy_length/src/Plugin/PasswordConstraint/PasswordLength.php b/password_policy_length/src/Plugin/PasswordConstraint/PasswordLength.php
index 2048a027..5c9607bd 100644
--- a/password_policy_length/src/Plugin/PasswordConstraint/PasswordLength.php
+++ b/password_policy_length/src/Plugin/PasswordConstraint/PasswordLength.php
@@ -27,13 +27,13 @@ public function validate($password, UserInterface $user) {
     $validation = new PasswordPolicyValidation();
     switch ($configuration['character_operation']) {
       case 'minimum':
-        if (strlen($password) < $configuration['character_length']) {
+        if (mb_strlen($password) < $configuration['character_length']) {
           $validation->setErrorMessage($this->formatPlural($configuration['character_length'], 'Password length must be at least 1 character.', 'Password length must be at least @count characters.'));
         }
         break;

       case 'maximum':
-        if (strlen($password) > $configuration['character_length']) {
+        if (mb_strlen($password) > $configuration['character_length']) {
           $validation->setErrorMessage($this->formatPlural($configuration['character_length'], 'Password length must not exceed 1 character.', 'Password length must not exceed @count characters.'));
         }
         break;
diff --git a/password_policy_length/tests/src/Unit/PasswordLengthTest.php b/password_policy_length/tests/src/Unit/PasswordLengthTest.php
index a896fdce..dc0099d3 100644
--- a/password_policy_length/tests/src/Unit/PasswordLengthTest.php
+++ b/password_policy_length/tests/src/Unit/PasswordLengthTest.php
@@ -86,6 +86,12 @@ public function lengthDataProvider() {
         'PasswordPassword',
         FALSE,
       ],
+      [
+        2,
+        'minimum',
+        '£',
+        FALSE,
+      ],
     ];
   }

