diff --git a/src/PasswordPolicyValidationManager.php b/src/PasswordPolicyValidationManager.php
index 051a842..113bb5f 100644
--- a/src/PasswordPolicyValidationManager.php
+++ b/src/PasswordPolicyValidationManager.php
@@ -84,6 +84,7 @@ class PasswordPolicyValidationManager implements PasswordPolicyValidationManager
     $role_applies = $this->passwordPolicyStorage->getQuery()
       ->condition('roles.*', $this->currentUser->getRoles(), 'IN')
       ->condition('show_policy_table', TRUE)
+      ->accessCheck(TRUE)
       ->execute();
     return !empty($role_applies);
   }
@@ -98,6 +99,7 @@ class PasswordPolicyValidationManager implements PasswordPolicyValidationManager
     $current_user_roles[] = "authenticated";
     $role_applies = $this->passwordPolicyStorage->getQuery()
       ->condition('roles.*', $current_user_roles, 'IN')
+      ->accessCheck(TRUE)
       ->execute();
     return !empty($role_applies);
   }